docs/proposals/secret-manager-store-api.md

```diff
@@ -0,0 +1,69 @@
+# Secrets Store Proposal
+
+**Author**: Emmanuel Bakare (@emmanuel-kubermatic)
+
+**Status**: Draft proposal; prototype in progress.
+
+## Goals
+The proposed fix for the issue [6671](https://github.com/kubermatic/kubermatic/issues/6671) 
+
+- Define secret using Vault signed certificates to assist with SSH access
+
+## Non-Goals
+
+
+## Motivation and Background
+
+A kubermatic user can assign ssh keys to clusters at creation time via the machine controller through cloud-init, but the user ssh agent also assists with updating the keys at runtime via the usersshkeys secret.
```

Comment:
`The keys are updated mainly via the UserSSHKeys object in the master cluster, so not directly on the secrets. Once the list is updated in the master, users then can attach keys to user clusters which would then create secrets that have the keys in their data field. Then those secrets are read from the seed by a user ssh key agent that would always rewrite the ssh keys on each UserSSHKey update.`

This explains it a bit more, because the real canonical truth is always coming from the UserSSHKeys in the master. Would say to formulate the text above in your style and add it here.

Reply:
Done